### PR TITLE
[FIX] Refactor design matrix and contrast formula for the two-sample T-test example

### LINKED ISSUE #4400: Is there an error in the design matrix from the second level two sample test example?
Hello,
I'm trying to follow the tutorial to deal with two-sample t-test using nilearn, but I'm finding something a bit confused.

In the code:
https://github.com/nilearn/nilearn/blob/3559d80643bbc44f1408664447d552181bd7845d/examples/05_glm_second_level/plot_second_level_two_sample_test.py#L66-L78
and the document page: https://nilearn.github.io/dev/auto_examples/05_glm_second_level/plot_second_level_two_sample_test.html#sphx-glr-auto-examples-05-glm-second-level-plot-second-level-two-sample-test-py

the author conducted an unpaired two samples t-test on two groups of participants (vertical vs horizontal, 16 participants in each group, totaling 32 participants). In setting up the design matrix, the author assigned the vertical group as `1`, the horizontal group as `-1`, and did not include an intercept term：

```
condition_effect = np.hstack(([1] * n_subjects, [-1] * n_subjects))
unpaired_design_matrix = pd.DataFrame(
    condition_effect[:, np.newaxis], columns=["vertical vs horizontal"]
)
```

However, in my opinion, the design matrix should be assigned as:
```
condition_effect = np.hstack(([1] * n_subjects, [0] * n_subjects))
unpaired_design_matrix = pd.DataFrame(
    {"vertical vs horizontal": condition_effect, 
     "intercept": 1}
)
```
which assign the vertical group and horizontal group as 1 and 0, and add an intercept for the linear model. 

The two upon design matrix can be plotted as:

image

In my understanding, for a GLM model, the design matrix either excludes the intercept with 1, 0, and 0, 1 in conjunction with contrast matrix (like 1 -1) for final comparison (SPM style), or it needs to be designed with 1, 0, and the intercept term added. (https://imaging.mrc-cbu.cam.ac.uk/imaging/PrinciplesStatistics#Statistics:_the_simplest_case_-_a_regression)

Also, I compared the T map with the result from SPM12:

image

It shows that the spmT_0001.nii is exactly the same as the result from the second design matrix.

So I am wondering if there were some thing wrong with this example?

### PR DESCRIPTION
**Pull request by YCHuang0610**
Closes #4400 

Changes proposed in this pull request:

- Simplify the design matrix for paired and unpaired design from [1 -1] to [1 0].
- Fix the uncorrect design matrix for paired design by adding a interception column.

### CODE REVIEW

File: examples/05_glm_second_level/plot_second_level_two_sample_test.py [hunk 1]
```diff
@@ -60,26 +60,30 @@
 second_level_input = sample_vertical["cmaps"] + sample_horizontal["cmaps"]
 
 # %%
-# Next, we model the effect of conditions (sample 1 vs sample 2).
+# Next, we assign the subjects for vertical and horizontal checkerboard respectively.
 import numpy as np
 
-condition_effect = np.hstack(([1] * n_subjects, [-1] * n_subjects))
+vertical_subjects = np.hstack(([1] * n_subjects, [0] * n_subjects))
+horizontal_subjects = np.hstack(([0] * n_subjects, [1] * n_subjects))
```
Comment (bthirion) [inline examples/05_glm_second_level/plot_second_level_two_sample_test.py:67]: Sorry I should have been clearer, but when you do this you make the design matrix  rank deficient: the sum of these two regressors, is equal to the some of all the subject regressors. 
This means that the design matrix is no longer invertible, and some contrasts are not estimable. Normally, you should get a warning when you run that code.

**Review by bthirion — APPROVED** [review]
LGTM.

File: examples/05_glm_second_level/plot_second_level_two_sample_test.py [hunk 2]
```diff
@@ -63,18 +63,20 @@
...
+# The design matrix for the unpaired test needs to add an intercept,
 # For the paired test, we include an intercept for each subject.
 subject_effect = np.vstack((np.eye(n_subjects), np.eye(n_subjects)))
 subjects = [f"S{i:02d}" for i in range(1, n_subjects + 1)]
 
 # %%
 # We then assemble those into design matrices
-unpaired_design_matrix = pd.DataFrame(
-    condition_effect[:, np.newaxis], columns=["vertical vs horizontal"]
+unpaired_design_matrix = pd.DataFrame({
```
Comment (Remi-Gau) [inline examples/05_glm_second_level/plot_second_level_two_sample_test.py:76]: Black is complaining about the formatting here:
https://github.com/YCHuang0610/nilearn/actions/runs/8893790642/job/24420709786#step:5:99

This should be fixed by applying all our style checks and autofixes and committing the changes.

```bash
pip install tox
tox 
```

**Comment by YCHuang0610** [comment]
> Thanks for your contribution, @YCHuang0610! Just a couple final things before we merge this.
> 
> Since this is your first contribution to nilearn could you please:
> 
> 1. add yourself to the authors section of CITATION.cff
> 2. also add this change to the `nilearn/doc/changes/latest.rst`
> 
> See https://nilearn.github.io/stable/development.html#changelog for more details on how to do this. And don't hesitate to ping me here if needed.
> 
> Thanks!



> Thanks for your contribution, @YCHuang0610! Just a couple final things before we merge this.
> 
> Since this is your first contribution to nilearn could you please:
> 
> 1. add yourself to the authors section of CITATION.cff
> 2. also add this change to the `nilearn/doc/changes/latest.rst`
> 
> See https://nilearn.github.io/stable/development.html#changelog for more details on how to do this. And don't hesitate to ping me here if needed.
> 
> Thanks!

Thank you for the guidance! 
I will make sure to add myself to the authors section and include this change in the nilearn/doc/changes/latest.rst as requested.
Thank you again for the opportunity to contribute.

**Comment by man-shu** [comment]
doc builder was complaining about the example's label in `latest.rst`.
Let's see if this fixes it.

**Review by man-shu — APPROVED** [review]
All good now. Merging!

**Pull request merged**